api/src/Page/Shipment.php

```diff
@@ -3058,20 +3123,27 @@ function($package, $index) {return array("piecenumber" => $index+1, "licenseplat
     function _create_shipment_shipment_request($shipment, array $dewars): int
     {
 
-        // if (!is_null($shipment['EXTERNALSHIPPINGIDTOSYNCHROTRON'])) {
-        //     return $shipment['EXTERNALSHIPPINGIDTOSYNCHROTRON'];
-        // }
+        $shipping_id = (int) $shipment['SHIPPINGID'];
+
+        $token = md5(uniqid());
```

Comment:
Bit of a nitpick, but `uniqid` could be "guessed", since we're MD5 hashing it rather than encoding it with an asymmetric cryptographic key pair

In this case, I think `openssl_random_pseudo_bytes` might work better:

```suggestion
        $token = md5(openssl_random_pseudo_bytes(7));
```

(This still isn't the safest, since MD5 hashes have been know to be subject to collisions for a while now, but I guess it's in tune with PHP's docs!)